### PR TITLE
Module 4

### PR DESCRIPTION
**Pull request by polylearner**
Vagrant file work for Module 4 Project exercise.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                echo 'export PYENV_ROOT=\"$HOME/.pyenv\"' >> ~/.bash_profile
+                echo 'export PATH=\"$PYENV_ROOT/bin:$PATH\"' >> ~/.bash_profile
+                echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n  eval \"$(pyenv init -)\"\nfi' >> ~/.bash_profile
+            fi
+            echo 'About to setup venv for python'
+            #if [ ! -d \"~/venv/bin/activate\" ]; then
+                sudo apt-get --assume-yes install python3-venv
+                python3 -m venv ~/venv
+                echo 'eval \"$(pyenv init -)\"' >> ~/.bash_profile
+                source ~/.profile
```
Comment (Jonesey13) [inline Vagrantfile]: But we've written our commands to `~/.bash_profile`?

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            if [ ! -d \"~/.pyenv\" ]; then
+                git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+                echo 'export PYENV_ROOT=\"$HOME/.pyenv\"' >> ~/.bash_profile
+                echo 'export PATH=\"$PYENV_ROOT/bin:$PATH\"' >> ~/.bash_profile
+                echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n  eval \"$(pyenv init -)\"\nfi' >> ~/.bash_profile
+            fi
+            echo 'About to setup venv for python'
+            #if [ ! -d \"~/venv/bin/activate\" ]; then
+                sudo apt-get --assume-yes install python3-venv
+                python3 -m venv ~/venv
```
Comment (Jonesey13) [inline Vagrantfile]: Why do we need these lines when python3 comes with `venv` anyway?

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                echo 'eval \"$(pyenv init -)\"' >> ~/.bash_profile
+                source ~/.profile
+            #fi
+                
+            echo 'Source venv'
+            source ~/venv/bin/activate
```
Comment (Jonesey13) [inline Vagrantfile]: Again we're using python before we're installed python!

File: Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            pyenv version
+            pyenv install 3.8.1
+            pyenv global 3.8.1
+            pyenv rehash
+
+            pip install --upgrade pip
```
Comment (Jonesey13) [inline Vagrantfile]: Our package manager is poetry and not pip

**Review by Jonesey13 — CHANGES_REQUESTED** [review]
This is nearly there but needs a couple of additions/changes:

- Please document how to run the VM in your `README.md` file
- Please add a trigger stage to your vagrantfile
- The script breaks trying to run pyenv atm (although it skips past it and uses the prepackaged python 3.6 instead thanks to use of `venv`).

**Review by JackMeadDev — APPROVED** [review]
Thanks for making those changes, I'm happy for you to merge this now as is.

**Pull request merged**